### PR TITLE
👽 Connect EditionSelect to the new API

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2023-06-01 上午9 25 29

### CODE REVIEW

**Review by nwingt — APPROVED** [review]

File: src/pages/nft/class/_classId/index.vue [hunk 1]
```diff
@@ -418,7 +419,10 @@ export default {
       this.updateNFTHistory();
       this.lazyFetchLIKEPrice();
       this.fetchUserCollectedCount();
-      const blockingPromises = [this.fetchISCNMetadata()];
+      const blockingPromises = [
+        this.fetchISCNMetadata(),
+        this.fetchNFTBookPriceByClassId(this.classId),
```
Comment (williamchong) [inline src/pages/nft/class/_classId/index.vue:424]: So we try to fetch a `book price` for all NFT? Can we safely assume we only have to fetch for those under a `NFT book` collection?

File: src/pages/nft/class/_classId/index.vue [hunk 2]
```diff
@@ -577,9 +581,17 @@ export default {
...
-      // TODO: Collect different edition
-      this.handleCollectFromPriceSection();
+    handleCollectFromEdition(selectedValue) {
+      logTrackerEvent(this, 'NFT', 'NFTCollect(Edition)', this.classId, 1);
+      if (this.getNFTBookStorePricesByClassId(this.classId)) {
+        const link = getNFTBookPurchaseLink({
+          classId: this.classId,
+          priceIndex: selectedValue,
+        });
+        window.open(link, '_blank');
```
Comment (williamchong) [inline src/pages/nft/class/_classId/index.vue:591]:
```suggestion
        window.open(link, '_blank', 'noopener');
```

File: src/mixins/nft.js [hunk 3]
```diff
@@ -293,17 +294,23 @@ export default {
...
+            stock: edition.stock,
+          }))
+        : [
+            {
+              name: 'Standard Edition',
+              priceLabel: this.formattedNFTPriceInLIKE,
+              value: 'standard',
+              stock: this.nftIsCollectable ? 500 : 0,
+            },
+          ];
```
Comment (williamchong) [inline src/mixins/nft.js:312]: If there is no result from `prices` we should just return empty data then disable any UI

File: src/pages/nft/class/_classId/index.vue [hunk 4]
```diff
@@ -577,9 +581,17 @@ export default {
       );
       return this.handleCollect();
     },
-    handleCollectFromEdition() {
-      // TODO: Collect different edition
-      this.handleCollectFromPriceSection();
+    handleCollectFromEdition(selectedValue) {
+      logTrackerEvent(this, 'NFT', 'NFTCollect(Edition)', this.classId, 1);
+      if (this.getNFTBookStorePricesByClassId(this.classId)) {
```
Comment (williamchong) [inline src/pages/nft/class/_classId/index.vue:586]: Maybe we should add a guard for sold out item

File: src/util/api/index.js [hunk 5]
```diff
@@ -361,3 +362,16 @@ export const postFollowCreator = ({ wallet, creator }) =>
   `/api/v2/users/${wallet}/followers?creator=${creator}`;
 export const getUserNotificationSettingsUrl = () =>
   `/api/v2/users/notification`;
+
+export const getNFTBookStorePricesByClassId = classId =>
+  `${LIKECOIN_API_BASE}/likernft/book/store/${classId}`;
+
+export const getNFTBookPurchaseLink = ({ classId, priceIndex }) => {
+  const qsPayload = {
+    form: NFT_BOOK_PLATFORM_LIKER_LAND,
```
Comment (williamchong) [inline src/util/api/index.js:371]:
```suggestion
    from: NFT_BOOK_PLATFORM_LIKER_LAND,
```

File: src/constant/index.js [hunk 6]
```diff
@@ -196,3 +196,5 @@ export const NFT_DISPLAY_STATE = {
 
 export const EMAIL_REGEX_STRING =
   '^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,4}$';
+
+export const NFT_BOOK_PLATFORM_LIKER_LAND = 'liker_land';
```
Comment (williamchong) [inline src/constant/index.js:200]: If we expect people to also use likerland as book sales page, we should accept `from` from query string. So this should only act as default value.

**Review by williamchong — CHANGES_REQUESTED** [review]

**Comment by nwingt** [comment]
@AuroraHuang22 #1118 is merged, please merge the update & fix on `<NFTEditionSelect/>`

File: src/util/api/index.js [hunk 7]
```diff
@@ -366,9 +366,13 @@ export const getUserNotificationSettingsUrl = () =>
 export const getNFTBookStorePricesByClassId = classId =>
   `${LIKECOIN_API_BASE}/likernft/book/store/${classId}`;
 
-export const getNFTBookPurchaseLink = ({ classId, priceIndex }) => {
+export const getNFTBookPurchaseLink = ({
+  classId,
+  priceIndex,
+  platform = NFT_BOOK_PLATFORM_LIKER_LAND,
```
Comment (williamchong) [inline src/util/api/index.js:372]: Please allow nft page to pass value into this param via its query string

**Review by williamchong — APPROVED** [review]

**Pull request merged**